src/dom_tree/validation.rs

```diff
@@ -0,0 +1,219 @@
+// Validation methods
+use super::tree::Tree;
+
+impl Tree {
+    /// Validates the structural integrity of the tree.
+    ///
+    /// This function checks for:
+    ///
+    /// 1. **Root Uniqueness**:
+    ///    - Ensures exactly one root node exists (`parent == None`) and it is `NodeId(0)`.
+    ///
+    /// 2. **Parent-Child Link Consistency**:
+    ///    - `first_child.parent == Self`
+    ///    - `first_child.prev_sibling == None`
+    ///    - `last_child.parent == Self`
+    ///    - `last_child.next_sibling == None`
+    ///    - All children in the chain from `first_child` to `last_child` have correct `parent`, `prev_sibling`, and `next_sibling` references.
+    ///
+    /// 3. **Sibling Link Consistency**:
+    ///    - `prev_sibling.next_sibling == Self`
+    ///    - `next_sibling.prev_sibling == Self`
+    ///
+    /// 4. **Valid Node References**:
+    ///    - All `NodeId` references must point to valid nodes within the tree.
+    ///
+    /// 5. **Orphaned Nodes** (when `check_orphans` is `true`):
+    ///   - Nodes with `parent == None` that are not the root node are considered orphaned.
+    ///
+    /// 6. **Cycle Detection**:
+    ///    - No cycles in parent chains (traverse `parent` links).
+    ///    - No cycles in sibling chains (traverse `next_sibling` links).
+    ///
+    /// Parameters:
+    /// - `check_orphans`: If `true`, check for orphaned nodes (nodes with `parent == None` that are not the root).
```

Comment:
Ooops, looks like some old code-comments from my own work got left in (my mistake!). 

vlidate() used to accept a parameter called "check_orphans", but I since removed it, so this Parameters section should be removed.